R/pfLineartBS.R

```diff
@@ -1,8 +1,55 @@
-
-pfLineartBS<- function(data, particles=1000, plot=FALSE, onlinePlot) {
+#' Particle Filter Example.
+#'
+#' @description The \code{pfLineartBS} function provides a simple example for
+#' \pkg{RcppSMC}. It is based on the first example in \code{SMCTC} and
+#' the discussion in Section 5.1 of Johansen (2009). A simple 'vehicle
+#' tracking' problem of 100 observations is solved with 1000 particles.
+#' 
+#' @param dat	A two-column matrix or dataframe containing x and y
+#' values. The default data set from Johansen (2009) is used as the
+#' default if no data is supplied.
+#' 
+#' @param particles	An integer specifying the number of particles.
+#' 
+#' @param plot	A boolean variable describing whether plot should
+#' illustrate the estimated path along with the data.
+#' 
+#' @param onlinePlot	A user-supplied callback function which is called with the
+#' x and y position vectors during each iteration of the algorithm; see
+#' pfExOnlinePlot for a simple example.
+#' 
+#' @return	The \code{pfLineartBS} function returns a \code{data.frame} containing as many rows as in
+#' the input data, and four columns corresponding to the estimated \eqn{x}{x} and
+#' \eqn{y}{y} coordinates as well as the estimated velocity in these two
+#' directions.
+#'
+#' @details The \code{pfLineartBS} function provides a simple example for
+#' \pkg{RcppSMC}. The model is linear with t-distributed innovations.
+#' It is based on the \code{pf} example in the
+#' \code{SMCTC} library, and discussed in the Section 5.1 of his
+#' corresponding paper (Johansen, 2009).
+#' 
+#' @references	A. M. Johansen. SMCTC: Sequential Monte Carlo in C++.
+#' Journal of Statistical Software, 30(6):1-41, April
+#' 2009. \url{http://www.jstatsoft.org/v30/i06/paper}
+#' 
+#' @seealso The SMCTC paper and code at \url{http://www.jstatsoft.org/v30/i06/paper}.
+#' 
+#' @examples
+#' res <- pfLineartBS(plot=TRUE)
+#' if (interactive()) ## if not running R CMD check etc
+#' res <- pfLineartBS(onlinePlot=pfLineartBSOnlinePlot)
+#' 
+#' @author		Adam M. Johansen, Dirk Eddelbuettel and Leah F. South
+#' @keywords	programming
+#' @export
+pfLineartBS<- function(dat, particles=1000, plot=FALSE, onlinePlot) {
 
     # if no data supplied, use default
-    if (missing(data)) data <- getPfLineartBSData()
+    if (missing(dat)){
```

Comment:
And now I see that the gung ho approach was entirely due to Dirk and I, but I think the comment remains valid. It's fine for these toy examples to make it easy for the user to see pretty pictures but we should avoid this with any code that might get used for anything important...